CodeJam.Main/Threading/TaskHelper.WithExtensions.cs

```diff
@@ -254,5 +267,139 @@ await taskFactory(timeoutOrCancellation.Token)
 				},
 				cancellation);
 		}
+
+		#region Internal implementation
+		/// <summary>
+		/// Internal implementation of <see cref="WithTimeout(Task,System.TimeSpan,Func{CancellationToken, Task},CancellationToken)"/> when timeout is not infinite.
+		/// </summary>
+		/// <param name="task">The task.</param>
+		/// <param name="timeout">The timeout.</param>
+		/// <param name="timeoutCallback">Callback that will be called on timeout.</param>
+		/// <param name="cancellation">The cancellation.</param>
+		private static async Task DoWithTimeout(
```

Comment:
I'd use Core suffix, `WaitWithTimeoutCore` or something like this.

Reply:
Also, as a rule of thumb, we do not add xml doc comments on private methods.
I think we can remove comments here